Replace workflow `release-starter` with updated `releaser.py` (#4586)
Scille/parsec-cloud: docs/development/release.md

Release candidate versions must have the naming ``v2.9.0-rc1``, ``v2.9.0-rc2`` etc.

### Create the release

> We have the script `misc/releaser.py` that provide an automated workflow to generate a new release/build.

Reviewer: ```suggestion
The script `misc/releaser.py` provides an automated workflow to generate a new release/build.
```

Note: The commits and release tag are annotated & signed, so you must have your GPG key
at the ready (and this key should be configured in your github account).

To create a new release, simply execute:

```shell
git fetch
python misc/releaser.py build --base=origin/master v2.9.0
```

> `git fetch` is used to update the remote ref, that is used to get the latest remote change on `master`.

> > We specify `--base=origin/master` because you are not enforce to be on the `master` branch to use that script.

Reviewer: ```suggestion
> `--base=origin/master` is used because you do not need to be on the `master` branch to use the script.
```

Author: I've applied the suggestion from @Ironicbay before you propose for that one, could re-review that part ?

The script will:

> - Ensure we specify a version that is greater than the current one.
> - Ensure we are on a clean _git env_ (no changes to be commited).

Reviewer: ```suggestion
- Ensure the release version is greater than the current version.
- Ensure _git env_ is clean (no changes to be commited).
```

- Create the release branch `releases/2.9` (and switch to it).
- Update the license Date & version.
- Update the parsec version across different files in our repository.
- Update `HISTORY.rst` with a new block generated using the news fragments found in `newsfragments/`.

>   At this step, the script will ask you to review the changes made so far (mostly to check if we correctly generated the [`HISTORY.rst`](/HISTORY.rst)).

Reviewer: ```suggestion
  At this step, the script will ask you to review the changes made so far (mostly to check if [`HISTORY.rst`](/HISTORY.rst) was correctly generated).
```

>   For instance, if some release candidate versions has been released before us,
>   now is the time to merge all those release notes into ours.

Reviewer: ```suggestion
  For instance, if some release candidate versions have been released before this one,
  now is the time to merge back the corresponding release notes.
```

- Create the release commit with the message `Bump version v2.8.1+dev -> v2.9.0`.

  The commit will contains the change made to `HISTORY.rst`, the various files referencing the parsec version, the updated license and the removed news fragments.

- Create the release tag `v2.9.0`.

>   The script will display the tag information, we need to review it before continuing (look at the tag signature).

Reviewer: ```suggestion
  The script will display the tag information, it needs to be reviewed before proceeding (check the tag signature).
```

- Create the dev commit with the message `Bump version v2.9.0 -> v2.9.0+dev`.

- Push the branch `releases/2.9` & the tag `v2.9.0`

  The script will require confirmation before continuing.

### Sign the installers

> ⚠️ This step is outdated with what we want to do.

Reviewer: ```suggestion
⚠️ This step is outdated.
```

On top of that the Python wheel of the project is going to be uploaded to Pypi.

### Create the release on Github

> ⚠️ This step is outdated with what we want to do.

Reviewer: ```suggestion
⚠️ This step is outdated.
```

> that doesn't contain an installer for there platform. Hence it's safe to create
> a new github release without any installer.

### Merge (or update) the Pull Request

> ⚠️ This step is outdated with what we want to do.

Reviewer: ```suggestion
⚠️ This step is outdated.
```

In the following we will consider we want to release version ``v2.9.1``.

### (Re)create the version branch

> If the release branch ``releases/2.9`` used for the ``2.9.0`` release has been
> removed, we must recreate it.

Reviewer: ```suggestion
If the release branch ``releases/2.9`` used for the ``2.9.0`` release has been
removed, it needs to be recreated.
```

> In case the release branch ``releases/2.9`` used to do ``2.9.0`` release has been
> removed, we must recreate it.

Reviewer 2: ```suggestion
If the release branch ``releases/2.9`` used for the ``2.9.0`` release has been
removed, we must recreate it.
```
